### PR TITLE
OptionsUI: Move internal options editors out of @grafana/ui

### PR DESCRIPTION
**Pull request by ryantxu**
A partial list of option editors is currently exposed in `@grafana/ui` -- these can not be used in any meaningful way, but it forces us to implement the core options editors within the UI package.

This PR moves the options editors out of the UI package so they can be implemented in grafana core.

This would let us avoid adding an unexported NumberInput to grafana UI in https://github.com/grafana/grafana/pull/46046

### CODE REVIEW

File: packages/grafana-ui/src/utils/standardEditors.tsx [hunk 1]
```diff
@@ -1,320 +1,17 @@
...
-import { StatsPickerEditor } from '../components/OptionsUI/stats';
-import { UnitValueEditor } from '../components/OptionsUI/units';
-import { Switch } from '../components/Switch/Switch';
-
 /**
  * Returns collection of common field config properties definitions
+ *
+ * @deprecated options editors are managed by grafana core
  */
 export const getStandardFieldConfigs = () => {
```
Comment (ryantxu) [inline packages/grafana-ui/src/utils/standardEditors.tsx:6]: I am not convinced that "deprecating" this is a good path.  The API is not really useable directly, so 🤷

Comment (ryantxu) [reply packages/grafana-ui/src/utils/standardEditors.tsx]: I updated the PR so we just remove them

**Review by torkelo — APPROVED** [review]

**Review by leventebalogh — APPROVED** [review]
LGTM 👍 

Nice that you have added the breaking changes info to the Levitate comment! Probably something that we should add to the tool as well.

**Comment by torkelo** [comment]
@ryantxu the backport for this failed, the question is do we need to backport this? I guess it would be nice to get rid of these from "cleaning up"  grafana/ui in v9.0 perspective

**Pull request merged**